sorting/odd_even_sort.c

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file
+ * @author [Edwin Ajong](https://github.com/eddybruv)
+ * @brief [Odd Even Sort](https://en.wikipedia.org/wiki/Odd%E2%80%93even_sort) implementation
+ * @details
+ * This algorithm is divided into two phases- Odd and Even Phase.
+ * The algorithm runs until the array elements are sorted and in each iteration two phases occurs- Odd and Even Phases.
+ * In the odd phase, we perform a bubble sort on odd indexed elements and in the even phase,
+ * we perform a bubble sort on even indexed elements.
+ * Time Complexity: O(N ^ 2)
+ */
+
+#include <assert.h>     /// for assert
+#include <stdbool.h>    /// for bool
+#include <stdio.h>      /// for IO operations
+#include <stdlib.h>     /// for dynammic memory allocation
+#include <time.h>       /// for random number generation
+
+/**
+ * @brief Swap numbers by reference(using pointers)
+ * @param first pointer to first number
+ * @param second pointer to second number
+ * @returns void
+ */
+
+void swap(int *first, int *second)
+{
+    int temp = *first;
+    *first = *second;
+    *second = temp;
+}
+
+/**
+ * @brief oddEvenSort sorts the array using the algorithm described above.
+ * @param arr array to be sorted
+ * @param size size of array
+ * @returns void
+ */
+
+void oddEvenSort(int *arr, int size)
+{
+    bool isSorted = false;
+    while(!isSorted)
+    {
+        isSorted = true;
+        int i;
+
+        //Even phase
+        for(i = 0; i <= size - 2; i += 2)
+        {
+            if(arr[i] > arr[i + 1])
+            {
+                swap(&arr[i], &arr[i + 1]);
+                isSorted = false;
+            }
+        }
+
+        //Odd phase
+        for(i = 1; i <= size - 2; i += 2)
+        {
+            if(arr[i] > arr[i + 1])
+            {
+                swap(&arr[i], &arr[i + 1]);
+                isSorted = false;
```

Comment:
Please add some documentation of what this code does here. 🙂

Reply:
@Panquesito7, I know the documentation maybe "amateur" but please it's my first time doing this, and it's quiet exiting.
Please take a look at what I've done and if there's any issue, I'll try my best to fix it.